src/utils/format.ts

```diff
@@ -1,10 +1,61 @@
 import BN from 'bn.js'
-import { fromWei } from 'web3-utils'
+import { TEN } from 'const'
 
-export function formatAmount(amount?: BN): string | null {
+const DEFAULT_DECIMALS = 4
+const DEFAULT_PRECISION = 18
+
+function _formatNumber(num: string): string {
+  return num.replace(/(\d)(?=(\d{3})+(?!\d))/g, '$1,')
+}
+
+function _decomposeBn(
+  amount?: BN,
+  amountPrecision = DEFAULT_PRECISION,
+  decimals = DEFAULT_DECIMALS,
+): { integerPart: BN; decimalPart: BN } {
+  // Discard the decimals we don't need
+  //  i.e. for WETH (precision=18, decimals=4) --> amount / 1e14
+  //        16.5*1e18 ---> 165000
+  const amountRaw = amount.divRound(TEN.pow(new BN(amountPrecision - decimals)))
```

Comment:
Just occurred to me. `amount` is an optional input (line 12), correct? What happens here if it's `undefined`?

I guess it shouldn't be optional here.

Reply:
Good point. We ensure that is not undefined before, but it should be part of the signature as you suggest